### PR TITLE
feat: Implement OpenTelemetry integration for Hertz

### PR DESCRIPTION
**Pull request by CoderPoet**
#### Provider
- [x] Out-of-the-box  default opentelemetry provider
- [x] Support setting via environment variables

#### Tracing
- [x] Support server and client hertz http tracing
- [x] Support automatic transparent transmission of peer service through meta info

#### Metrics
- [x] Support hertz http metrics [R.E.D]
- [x] Support service topology map metrics [Service Topology Map]
- [x] Support go runtime metrics

#### Logging
- [x] Extend hertz logger based on logrus
- [x] Implement tracing auto associated logs

## Configuration via environment variables
- [Exporter](https://opentelemetry.io/docs/reference/specification/protocol/exporter/)
- [SDK](https://opentelemetry.io/docs/reference/specification/sdk-environment-variables/#general-sdk-configuration)

### CODE REVIEW

**Comment by li-jin-gou** [comment]
如果用 `kitex` 的 `obs` 扩展声明了一个 `provider`,还需要在 `hertz` 中声明嘛

**Comment by CoderPoet** [comment]
> 如果用 `kitex` 的 `obs` 扩展声明了一个 `provider`,还需要在 `hertz` 中声明嘛

所以做成 sub module 了，哈哈，用户可选

File: version.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package hertz
+
+// Name and Version info of this framework, used for statistics and debug
+const (
+	Name    = "Hertz"
```
Comment (Duslia) [inline version.go:21]: 这个换成  obs-opentelemetry 的版本吧

Comment (CoderPoet) [reply version.go:21]: 好滴

**Review by Duslia — APPROVED** [review]

**Pull request merged**